digest/digest.go

```diff
@@ -123,6 +123,20 @@ func (d Digest) Algorithm() Algorithm {
 	return Algorithm(d[:d.sepIndex()])
 }
 
+// Verifier returns a writer object that can be used to verify a stream of
+// content against the digest. If the digest is invalid, an error will be
+// returned.
+func (d Digest) Verifier() (Verifier, error) {
+	if err := d.Validate(); err != nil {
```

Comment:
Do you think we still need to validate first? The fear before was people would call `NewDigestVerifier` by typecasting a string to `digest.Digest`, but `digest.Digest("some content").Verifier()` is a less likely pattern.